### PR TITLE
Rework defaults merging logic

### PR DESCRIPTION
**Pull request by sgallagher**
Fixes: https://github.com/fedora-modularity/libmodulemd/issues/378

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

### CODE REVIEW

File: modulemd/modulemd-defaults.c [hunk 1]
```diff
@@ -755,156 +764,250 @@ modulemd_defaults_merge (ModulemdDefaults *first,
...
+                                                ModulemdIntent *into_intent,
+                                                const gchar *intent_name,
+                                                guint64 from_modified,
+                                                guint64 into_modified)
+{
+  const gchar *from_default_stream = NULL;
+  const gchar *into_default_stream = NULL;
+
+  g_return_if_fail (from_intent && MODULEMD_IS_INTENT (from_intent));
+  g_return_if_fail (from_intent && MODULEMD_IS_INTENT (into_intent));
```
[the remark this answers is not in the document]
Comment (sgallagher) [reply modulemd/modulemd-defaults.c]: Good catch. That was a copy-paste mistake.

File: modulemd/modulemd-defaults.c [hunk 2]
```diff
@@ -755,156 +764,250 @@ modulemd_defaults_merge (ModulemdDefaults *first,
...
-                           "Conflicting profile defaults when merging "
-                           "defaults for module %s",
-                           modulemd_defaults_peek_module_name (first));
-              return NULL;
+              modulemd_defaults_set_default_stream (merged,
+                                                    second->default_stream);
+            }
+          else if (first->modified == second->modified)
+            {
+              if (g_strcmp0 (first->default_stream, second->default_stream))
```
[the remark this answers is not in the document]
Comment (sgallagher) [reply modulemd/modulemd-defaults.c]: I think that may be a relic of some reorganization of the code I did there. You're right, it doesn't need to protect against NULL, so using the more readable function name makes sense. Thanks.

**Comment by sgallagher** [comment]
Regarding the branch name, I was working on two things at the same time and must have gotten my wires crossed when I named this one. Ah well.

File: modulemd/modulemd-defaults.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgallagher) [reply modulemd/modulemd-defaults.c]: That also makes me aware that I missed doing the NULL-check at line 756 also. Fixed.

**Comment by mmathesius** [comment]
LGTM. Thanks!

**Pull request merged**